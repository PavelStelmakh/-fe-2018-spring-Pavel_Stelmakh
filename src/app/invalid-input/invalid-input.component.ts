import { Component, OnInit, Input } from '@angular/core';

@Component({
  selector: 'app-invalid-input',
  templateUrl: './invalid-input.component.html',
  styleUrls: ['./invalid-input.component.scss']
})
export class InvalidInputComponent {
  @Input() message: string;
  @Input() type: ['error', 'success'];

  constructor() { }

}
