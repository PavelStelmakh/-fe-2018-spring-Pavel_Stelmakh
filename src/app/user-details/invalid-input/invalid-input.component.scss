$background_error: #fa5655;
$background_success: #00c968;

:host {
    display: block;
    position: absolute;
    width: 450px;
    margin-left: 25px;
    margin-top: 5px;
    font-family: Verdana;
}

div {
    height: 50px;        
    line-height: 25px;
    color: rgba(0, 0, 0, 0.7);
    border-bottom-right-radius: 15px;
    border-bottom-left-radius: 15px;
}

.success {
    background-color: $background_success;    
    line-height: 50px;
}

.error {
    background-color: $background_error;
}