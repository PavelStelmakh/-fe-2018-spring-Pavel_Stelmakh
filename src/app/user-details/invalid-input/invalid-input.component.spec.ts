import { TestBed, async, ComponentFixture } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { InvalidInputComponent } from './invalid-input.component';

describe('InvalidInputComponent', () => {
  let fixture: ComponentFixture<InvalidInputComponent>;
  let component: InvalidInputComponent;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        InvalidInputComponent
      ],
      schemas: [
       NO_ERRORS_SCHEMA
      ]
    }).compileComponents();

  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InvalidInputComponent);
    component = fixture.componentInstance;
  });

  it('renders markup to snapshot', () => {
    component.message = 'Username must consist of one or two word with camel case sensitive';
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

});
