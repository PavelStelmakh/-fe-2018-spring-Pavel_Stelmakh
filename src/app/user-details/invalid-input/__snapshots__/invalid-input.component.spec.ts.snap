// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvalidInputComponent renders markup to snapshot 1`] = `
<app-invalid-input
  message={[Function String]}
>
  <div
    class="undefined"
  >
     Username must consist of one or two word with camel case sensitive
  
  </div>
</app-invalid-input>
`;
